lusidtools/cocoon/cocoon_printer.py

```diff
@@ -0,0 +1,171 @@
+import pandas as pd
+import numpy as np
+
+
+def check_dict_for_required_keys(target_dict: dict, target_name: str, required_keys: list):
+    """
+    This function checks that a named dictionary contains a list of required keys
+    :param dict target_dict: Dictionary to check for keys in
+    :param str target_name: Variable name of dictionary
+    :param list required_keys: List of required keys
+    :return:
+    """
+    for key in required_keys:
+        if key not in target_dict.keys():
+            raise ValueError(f"'{key}' missing from {target_name}. {target_name} must include {required_keys}"
+                             f"as keys")
+
+
+def get_errors_from_response(list_of_API_exceptions):
+    """
+    This function gets the status code and reason from API exception
+    :param list list_of_API_exceptions:
+    :return pd.DataFrame: a Pandas DataFrame containing the reasons and status codes for ApiExceptions from a request
+    """
+    # get API exception
+
```

Comment:
Check here that every item in your list is a lusid.exceptions.ApiException so you don't get an error later.